numpyro/util.py

```diff
@@ -408,3 +408,121 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
         lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
     )
     return tree_map(lambda y: jnp.reshape(y, batch_shape + jnp.shape(y)[1:]), ys)
+
+
+def format_shapes(trace, *, title="Trace Shapes:", last_site=None, log_prob=False):
+    """
+    Returns a string showing a table of the shapes of all sites in the
+    trace.
+
+    :param dict trace: The model trace to format. Use ``numpyro.handlers.trace`` to
+        produce the trace.
```

Comment:
How about rephrasing the last sentence to: "Use :class:`~numpyro.handlers.trace` handler (or funsor :class:`~numpyro.contrib.funsor.enum_messenger.trace` handler for enumeration) to produce the trace."